cli/src/template.pest

```diff
@@ -89,10 +92,21 @@ formal_parameters = {
   | ""
 }
 
+// NOTE: string pattern identifiers additionally allow "-" in them, which
+// results in some oddness with the `-` operator, though does not yet cause
+// ambiguity. This may prove annoying at some future point.
+string_pattern_identifier = @{ (ASCII_ALPHA | "_") ~ (ASCII_ALPHANUMERIC | "_" | "-")* }
+string_pattern = {
```

Comment:
Since I've merged https://github.com/jj-vcs/jj/pull/7380, we'll probably need `${ .. }`. Can you also add a test for `prefixed-pattern : "with space"`, which should be a parse error?